### PR TITLE
feat: implement named stoichiometry and advanced event options for AntimonySerializer

### PR DESCRIPTION
**Pull request by dyrpsf**
### Description
This PR acts as a direct follow-up to the Phase 1-3 implementation merged in #305. It addresses the advanced edge cases and feature requests highlighted by @luciansmith during the initial review, ensuring the serializer can handle complex SBML models cleanly and token-efficiently.

### Key Updates
* **Named Stoichiometry:** Added logic to `toAntimony(Reaction r)` to check for and serialize named stoichiometries (e.g., dynamically mapping to `n S2 => S3` if the `SpeciesReference` has an ID).
* **Advanced Event Options:** Expanded `toAntimony(Event e)` to support complex event configurations:
  * Delays (`delay=...`)
  * Priorities (`priority=...`)
  * Initial Values (`t0=false`)
  * Persistence (`persistent=false`)
* **Token Efficiency:** The serializer strictly respects SBML defaults. Options like `t0` and `persistent` are only appended if explicitly set to `false`, keeping the resulting Antimony string as concise as possible.

### Testing
Expanded `AntimonySerializerTest.java` to cover these new edge cases. All tests are passing cleanly (`BUILD SUCCESS`).

With these advanced features in place, the underlying serialization logic is fully complete and ready to be adapted into the upcoming `TreeWalk` Visitor pattern we are discussing in #262!

### CODE REVIEW

**Comment by dyrpsf** [comment]
@draeger, @luciansmith - Just pinging you both here since this PR specifically implements the advanced edge cases we discussed at the end of #305! 

Whenever you have a moment to review the logic for the named stoichiometries and event formatting, let me know if it looks good or if any tweaks are needed!

**Comment by luciansmith** [comment]
That's not actually the correct syntax for delays! It's:

at S1-5 after B>3: x = 4

i.e. '[formula] after'

Are you ensuring that the produced Antimony is correct by trying to parse it with libAntimony?  And then ensuring that libAntimony gives you essentially the same SBML when round-tripped?  It might help with things like this.

**Comment by dyrpsf** [comment]
> That's not actually the correct syntax for delays! It's:
> 
> at S1-5 after B>3: x = 4
> 
> i.e. '[formula] after'
> 
> Are you ensuring that the produced Antimony is correct by trying to parse it with libAntimony? And then ensuring that libAntimony gives you essentially the same SBML when round-tripped? It might help with things like this.

Ah, thank you so much for the correction! I completely misunderstood the Antimony documentation regarding the delay syntax. `at [delay] after [trigger]` makes perfect sense. 

Regarding your question about round-trip testing with `libAntimony`: that is a fantastic suggestion. Currently, the JUnit test suite in JSBML is purely doing string/regex matching against expected literals since JSBML is a pure Java environment. However, moving forward, I will definitely use the Python `libantimony` bindings locally to manually validate these complex strings and ensure they parse and round-trip correctly!

I will push a fix for the delay syntax right away.

**Comment by dyrpsf** [comment]
@luciansmith Fix pushed! The tests are passing cleanly with the updated `at [delay] after [trigger]` syntax. Thanks again for catching that and for the tip on using `libAntimony` for round-tripping!

File: core/src/org/sbml/jsbml/util/AntimonySerializer.java [hunk 1]
```diff
@@ -178,7 +178,11 @@ public static String toAntimony(Reaction r) {
         // 2. Reactants
         for (int i = 0; i < r.getReactantCount(); i++) {
             SpeciesReference sr = r.getReactant(i);
-            if (sr.isSetStoichiometry() && sr.getStoichiometry() != 1.0) {
+            
+            // Check for Named Stoichiometry (e.g., 'n S2')
+            if (sr.isSetId()) {
+                ant.append(sr.getId()).append(" ");
+            } else if (sr.isSetStoichiometry() && sr.getStoichiometry() != 1.0) {
```
Comment (draeger) [inline core/src/org/sbml/jsbml/util/AntimonySerializer.java]: Very minor comment but I usually write `1d` rather than `1.0` because the latter could also be interpreted as a float value. Similarly, I'd use `1f` and `1L` to make clear what is meant.

File: core/src/org/sbml/jsbml/util/AntimonySerializer.java [hunk 2]
```diff
@@ -250,11 +258,32 @@ public static String toAntimony(Event e) {
...
 
-        ant.append("at (");
-        if (e.isSetTrigger() && e.getTrigger().isSetMath()) {
+        ant.append("at ");
+        boolean hasTrigger = e.isSetTrigger() && e.getTrigger().isSetMath();
+        boolean hasDelay = e.isSetDelay() && e.getDelay().isSetMath();
+
+        if (hasDelay && hasTrigger) {
+            ant.append(ASTNode.formulaToString(e.getDelay().getMath()));
+            ant.append(" after ");
```
Comment (draeger) [inline core/src/org/sbml/jsbml/util/AntimonySerializer.java:267]: Also, it can make sense to collect Antimony keywords as `public static final String` values in some interface that we can then implement to inherit all the constants directly.

File: core/src/org/sbml/jsbml/util/AntimonySerializer.java [hunk 3]
```diff
@@ -250,11 +258,32 @@ public static String toAntimony(Event e) {
...
-        ant.append("): ");
+
+        // Advanced Event Options
+        if (e.isSetPriority() && e.getPriority().isSetMath()) {
+            ant.append(", priority=").append(ASTNode.formulaToString(e.getPriority().getMath()));
+        }
+        if (e.isSetTrigger()) {
+            org.sbml.jsbml.Trigger t = e.getTrigger();
+            if (t.isSetInitialValue() && !t.getInitialValue()) {
+                ant.append(", t0=false");
```
Comment (draeger) [inline core/src/org/sbml/jsbml/util/AntimonySerializer.java]: Can spaces be added in the output, i.e., `t0 = false`?

File: core/src/org/sbml/jsbml/util/AntimonySerializer.java [hunk 4]
```diff
@@ -250,11 +258,32 @@ public static String toAntimony(Event e) {
...
+            }
+            if (t.isSetPersistent() && !t.getPersistent()) {
+                ant.append(", persistent=false");
```
Comment (draeger) [inline core/src/org/sbml/jsbml/util/AntimonySerializer.java]: Here also blanks please.

**Review by draeger — CHANGES_REQUESTED** [review]
A few minor change requests about formatting only.

File: core/src/org/sbml/jsbml/util/AntimonySerializer.java (hunk 1 above)
Comment (dyrpsf) [reply core/src/org/sbml/jsbml/util/AntimonySerializer.java]: Fixed! Updated to `1d` to ensure it is explicitly treated as a double.

File: core/src/org/sbml/jsbml/util/AntimonySerializer.java (hunk 3 above)
Comment (dyrpsf) [reply core/src/org/sbml/jsbml/util/AntimonySerializer.java]: Good catch, added the requested spacing here for better readability!

File: core/src/org/sbml/jsbml/util/AntimonySerializer.java (hunk 4 above)
Comment (dyrpsf) [reply core/src/org/sbml/jsbml/util/AntimonySerializer.java]: Spacing added here too!

**Review by draeger — APPROVED** [review]
I'll merge this now and we can add an interface collecting Antimony keywords later.

**Comment by dyrpsf** [comment]
Thank you for the merge, @draeger! 

Funny timing - I actually just finished building that exact `AntimonyConstants` interface locally right as you merged this! Since the code is already fully written and tested, I will go ahead and open a quick, separate PR for it now so we have that clean architecture locked in.

Update: new PR : #311 .

File: core/src/org/sbml/jsbml/util/AntimonySerializer.java (hunk 2 above)
Comment (dyrpsf) [reply core/src/org/sbml/jsbml/util/AntimonySerializer.java:267]: Done (PR #311).
